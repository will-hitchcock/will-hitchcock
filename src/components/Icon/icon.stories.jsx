import React from 'react'
import { storiesOf } from '@storybook/react'
import Icon from './index'

storiesOf('Icon', module)
  .add('linkedin', () => (
    <Icon name="linkedin" color="#0077B5" size={32} />
  ))
  .add('twitter', () => (
    <Icon name="twitter" color="#1DA1F2" size={64} />
  ))
  .add('github', () => (
    <Icon name="github" color="#000000" size={128} />
  ))
