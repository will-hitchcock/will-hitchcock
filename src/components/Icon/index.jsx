import React from 'react'
import { string, number, oneOf } from 'prop-types'
import { Icon as IconKit } from 'react-icons-kit'
import { linkedin } from 'react-icons-kit/fa/linkedin'
import { twitter } from 'react-icons-kit/fa/twitter'
import { github } from 'react-icons-kit/fa/github'
import styled from 'styled-components'

const Container = styled.div`
  display: inline-flex;
  color: ${(props) => props.color}
`

const getIcon = (name) => {
  switch (name) {
    case 'linkedin':
      return linkedin
    case 'twitter':
      return twitter
    case 'github':
      return github
    default:
      return null
  }
}

const Icon = ({ name, color, size }) => (
  <Container color={color}>
    <IconKit icon={getIcon(name)} size={size} />
  </Container>
)

Icon.propTypes = {
  name: oneOf([
    'linkedin',
    'twitter',
    'github',
  ]).isRequired,
  color: string.isRequired,
  size: number.isRequired,
}

export default Icon
