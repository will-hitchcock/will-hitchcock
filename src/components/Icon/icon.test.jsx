import React from 'react'
import { shallow } from 'enzyme'
import Icon from './index'

describe('Icon', () => {
  it('renders a linkedin <Icon />, color #0077B5, size 32', () => {
    expect(shallow(
      <Icon name="linkedin" color="#0077B5" size={32} />,
    )).toMatchSnapshot()
  })

  it('renders a twitter <Icon />, color #1DA1F2, size 64', () => {
    expect(shallow(
      <Icon name="twitter" color="#1DA1F2" size={64} />,
    )).toMatchSnapshot()
  })

  it('renders a github <Icon />, color #000000, size 128', () => {
    expect(shallow(
      <Icon name="twitter" color="#000000" size={128} />,
    )).toMatchSnapshot()
  })
})
