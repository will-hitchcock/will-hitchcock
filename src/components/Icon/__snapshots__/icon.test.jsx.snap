// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders a github <Icon />, color #000000, size 128 1`] = `
<styled.span
  color="#000000"
>
  <Icon
    fill="currentColor"
    icon={
      Object {
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M1620 408q-67 98-162 167 1 14 1 42 0 130-38 259.5t-115.5 248.5-184.5 210.5-258 146-323 54.5q-271 0-496-145 35 4 78 4 225 0 401-138-105-2-188-64.5t-114-159.5q33 5 61 5 43 0 85-11-112-23-185.5-111.5t-73.5-205.5v-4q68 38 146 41-66-44-105-115t-39-154q0-88 44-163 121 149 294.5 238.5t371.5 99.5q-8-38-8-74 0-134 94.5-228.5t228.5-94.5q140 0 236 102 109-21 205-78-37 115-142 178 93-10 186-50z",
            },
            "name": "path",
          },
        ],
        "viewBox": "0 0 1664 1792",
      }
    }
    size={128}
  />
</styled.span>
`;

exports[`Icon renders a linkedin <Icon />, color #0077B5, size 32 1`] = `
<styled.span
  color="#0077B5"
>
  <Icon
    fill="currentColor"
    icon={
      Object {
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M349 625v991h-330v-991h330zM370 319q1 73-50.5 122t-135.5 49h-2q-82 0-132-49t-50-122q0-74 51.5-122.5t134.5-48.5 133 48.5 51 122.5zM1536 1048v568h-329v-530q0-105-40.5-164.5t-126.5-59.5q-63 0-105.5 34.5t-63.5 85.5q-11 30-11 81v553h-329q2-399 2-647t-1-296l-1-48h329v144h-2q20-32 41-56t56.5-52 87-43.5 114.5-15.5q171 0 275 113.5t104 332.5z",
            },
            "name": "path",
          },
        ],
        "viewBox": "0 0 1536 1792",
      }
    }
    size={32}
  />
</styled.span>
`;

exports[`Icon renders a twitter <Icon />, color #1DA1F2, size 64 1`] = `
<styled.span
  color="#1DA1F2"
>
  <Icon
    fill="currentColor"
    icon={
      Object {
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M1620 408q-67 98-162 167 1 14 1 42 0 130-38 259.5t-115.5 248.5-184.5 210.5-258 146-323 54.5q-271 0-496-145 35 4 78 4 225 0 401-138-105-2-188-64.5t-114-159.5q33 5 61 5 43 0 85-11-112-23-185.5-111.5t-73.5-205.5v-4q68 38 146 41-66-44-105-115t-39-154q0-88 44-163 121 149 294.5 238.5t371.5 99.5q-8-38-8-74 0-134 94.5-228.5t228.5-94.5q140 0 236 102 109-21 205-78-37 115-142 178 93-10 186-50z",
            },
            "name": "path",
          },
        ],
        "viewBox": "0 0 1664 1792",
      }
    }
    size={64}
  />
</styled.span>
`;
